zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/publishing/legacycacheapi/LegacyCacheApiPayloadBuilder.java

```diff
@@ -0,0 +1,205 @@
+                String updatedURI = URIUtils.getNSegmentsAfterSegmentInput(uri, BULLETIN_SEGMENT, 1) + "/latest";
+                LegacyCacheApiPayload legacyCacheApiPayloadLatest = new LegacyCacheApiPayload(collectionId, updatedURI, clearCacheDate);
+                payloadsByPath.put(legacyCacheApiPayloadLatest.uriToUpdate, legacyCacheApiPayloadLatest);
+            }
+        }
+
+        private boolean isValid(LegacyCacheApiPayload payload) {
+            if (StringUtils.isEmpty(payload.uriToUpdate)) {
+                info().data("collectionId", payload.collectionId)
+                        .log("invalid payload: URI was empty or null");
+                return false;
+            }
+            return true;
+        }
+
+        public static String getCanonicalPagePath(String incomingUrl, String  collectionId) {
+            try {
+                if (StringUtils.isEmpty(incomingUrl)) {
+                    return incomingUrl;
+                }
+
+                String updatedUri = incomingUrl.trim();
+                if (StringUtils.isEmpty(updatedUri)) {
+                    return "";
+                }
+
+                // Order matters:
+                // 1. If starts with ons return original
+                if (updatedUri.startsWith(ONS_PREFIX)) {
+                    return updatedUri;
+                }
+
+                // 2. If starts with /visualisation/ then /visualisation/xxx/xxxx/ ---> /visualisation/xxx (keep 1 segment) - return here
+                if (updatedUri.startsWith(VISUALISATIONS_SEGMENT)) {
+                    return URIUtils.getNSegmentsAfterSegmentInput(updatedUri, VISUALISATIONS_SEGMENT, 1);
+                }
+
+                // 3. If resource then e.g. /embed?uri='/something' ---> /something:
+                //      /chartconfig
+                //      /chartimage
+                //      /embed
+                //      /chart
+                //      /resource
+                //      /generator
+                //      /file
+                //      /export
+                for (String keyword : RESOURCE_TYPE_SEGMENTS) {
+                    if (updatedUri.startsWith(keyword)) {
+                        updatedUri = URIUtils.getQueryParameterFromURL(updatedUri, "uri");
+                        break;
+                    }
+                }
+
+                // 4. If bulletin or article then /xxxx/bulletin/xxxx1/xxxx2/xxxx3 --> /bulletin/xxxx1/xxxx2  (keep 2 segment) & RETURN modified
+                if (updatedUri.contains(ARTICLE_SEGMENT)) {
+                    return URIUtils.getNSegmentsAfterSegmentInput(updatedUri, ARTICLE_SEGMENT, 2);
+                }
+                if (updatedUri.contains(BULLETIN_SEGMENT)) {
+                    return URIUtils.getNSegmentsAfterSegmentInput(updatedUri, BULLETIN_SEGMENT, 2);
+                }
+                // 5. If qmis or adhoc then /xxxx/qmis/xxx1/xxx2 --> /qmis/xxx1 & RETURNS modified
```

Comment:
Methodologies are missing.

You added tests for methodologies and your tests pass, which means that the tests are weak.